tfjs-backend-webgpu/src/kernels/from_pixels_webgpu.ts

```diff
@@ -26,35 +26,144 @@ export class FromPixelsProgram implements WebGPUProgram {
   variableNames = ['A'];
   dispatchLayout: {x: number[]};
   dispatch: [number, number, number];
+  pipeline: GPUComputePipeline;
+  bindGroupLayout: GPUBindGroupLayout;
+
+  uniform: GPUBuffer;
+  lastUniformContent = [0, 0, 0];
+
+  inputTexture: GPUTexture = null;
+  lastPixelSize = {width: 0, height: 0};
+
+  private disposed = false;
 
   constructor(outputShape: number[]) {
     this.outputShape = outputShape;
     this.dispatchLayout = flatDispatchLayout(this.outputShape);
     this.dispatch = computeDispatch(this.dispatchLayout, this.outputShape);
```

Comment:
Sorry that I didn't notice it earlier. But your dispatch should be like below. With this change, I think your shader should be accelerated a lot.
```
this.workGroupSize = [256, 1, 1]; // Maybe other values. Please do another round of experiments.
this.workPerThread = numChannels;
this.dispatch = computeDispatch(this.dispatchLayout, this.outputShape, this.workGroupSize, [this.workPerThread, 1 , 1]);

Reply:
Done.